src/static/js/medical.js

```diff
@@ -735,6 +806,40 @@ $(function() {
             } else {
                 $("#treatmentrulecalc").hide();
             }
+            medical.rebuild_treatment_times();
+        },
+
+        rebuild_treatment_times: function() {
+            const sm = $("#singlemulti").val();
+            const row = $("#treatmenttimesrow");
+            const table = row.closest("table");
+
+            table.find("tr.asm-tx-time-row").remove();
+
+            if (sm != medical.TREATMENT_MULTI && sm != medical.TREATMENT_SINGLE) {
+                row.hide();
+                return;
+            }
+
+            const perday = (sm == medical.TREATMENT_SINGLE) ? 1 : format.to_int($("#timingrule").val() || "1");
+            if (!perday || perday <= 0) {
+                row.hide();
+                return;
+            }
+
+            row.show();
+
+            let html = "";
+            for (let i = 1; i <= perday; i++) {
+                const t = medical.treatment_time_for({ TREATMENTNUMBER: i, TOTALTREATMENTS: perday }) || "";
+                html += '<tr class="asm-formfield asm-tx-time-row">' +
+                        '<td><label>' + _("Dose {0}").replace("{0}", i) + '</label></td>' +
+                        '<td><input type="text" class="asm-textbox asm-timebox" ' +
+                        'name="treatmenttime' + i + '" value="' + t + '"/></td>' +
+                        '</tr>';
```

Comment:
Use the tableform render functions instead of hand cranking the HTML, eg:

```
tableform.render_time({ name: "treatmenttime" + i, 
    value: t, 
    label: _("Dose {0}").replace("{0}", i), 
    rowclasses: "asm-tx-time-row });
```